lib/methods/ifwapichecktask.cpp

```diff
@@ -344,6 +249,31 @@ void IfwApiCheckTask::ScriptFunc(const Checkable::Ptr& checkable, const CheckRes
 	String username = resolveMacros("$ifw_api_username$");
 	String password = resolveMacros("$ifw_api_password$");
 
+	std::function<void(const Value& cmdLine, const ProcessResult&)> resultHandler;
```

Comment:
That `const ProcessResult&` seems to be unnecessary, none of the implementations uses it and it's always called with `ProcessResult{}`.

Also, `cmdLine` shouldn't really need special treatment here. That can also be set in the check result (and in case of using the `else` branch directly below, already is).

To me, it would make more sense if the function took the check result as the parameter, that's the common thing both variants are using and having a callback for "once done, call it with the result" sounds sane to me.